tests/unit/test_git_utils.py

```diff
@@ -0,0 +1,177 @@
+@patch('tempfile.mkdtemp', return_value=REPO_DIR)
+@patch('sagemaker.git_utils._validate_git_config')
+@patch('sagemaker.git_utils._checkout_branch_and_commit',
+       side_effect=subprocess.CalledProcessError(returncode=1, cmd='git checkout {}'.format(COMMIT)))
+@patch('os.path.isfile', return_value=True)
+@patch('os.path.isdir', return_value=True)
+@patch('os.path.exists', return_value=True)
+def test_git_clone_repo_commit_not_exist(exists, isdir, isfile, checkout_branch_and_commit,
+                                         validate_git_config, mkdtemp, check_call):
+    git_config = {'repo': GIT_REPO, 'branch': BRANCH, 'commit': COMMIT}
+    entry_point = 'entry_point'
+    source_dir = 'source_dir'
+    dependencies = ['foo', 'bar']
+    with pytest.raises(subprocess.CalledProcessError) as error:
+        git_utils.git_clone_repo(git_config, entry_point, source_dir, dependencies)
+    assert 'returned non-zero exit status' in str(error)
+
+
+@patch('subprocess.check_call')
+@patch('tempfile.mkdtemp', return_value=REPO_DIR)
+@patch('sagemaker.git_utils._validate_git_config')
+@patch('sagemaker.git_utils._checkout_branch_and_commit')
+@patch('os.path.isfile', return_value=False)
+@patch('os.path.isdir', return_value=True)
+@patch('os.path.exists', return_value=True)
+def test_git_clone_repo_entry_point_not_exist(exists, isdir, isfile, checkout_branch_and_commit,
+                                              validate_git_config, mkdtemp, check_call):
+    git_config = {'repo': GIT_REPO, 'branch': BRANCH, 'commit': COMMIT}
+    entry_point = 'entry_point_that_does_not_exist'
+    source_dir = 'source_dir'
+    dependencies = ['foo', 'bar']
+    with pytest.raises(ValueError) as error:
+        git_utils.git_clone_repo(git_config, entry_point, source_dir, dependencies)
+    assert 'Entry point does not exist in the repo.' in str(error)
+
+
+@patch('subprocess.check_call')
+@patch('tempfile.mkdtemp', return_value=REPO_DIR)
+@patch('sagemaker.git_utils._validate_git_config')
+@patch('sagemaker.git_utils._checkout_branch_and_commit')
+@patch('os.path.isfile', return_value=True)
+@patch('os.path.isdir', return_value=False)
+@patch('os.path.exists', return_value=True)
+def test_git_clone_repo_source_dir_not_exist(exists, isdir, isfile, checkout_branch_and_commit,
+                                             validate_git_config, mkdtemp, check_call):
+    git_config = {'repo': GIT_REPO, 'branch': BRANCH, 'commit': COMMIT}
+    entry_point = 'entry_point'
+    source_dir = 'source_dir_that_does_not_exist'
+    dependencies = ['foo', 'bar']
+    with pytest.raises(ValueError) as error:
+        git_utils.git_clone_repo(git_config, entry_point, source_dir, dependencies)
+    assert 'Source directory does not exist in the repo.' in str(error)
+
+
+@patch('subprocess.check_call')
+@patch('tempfile.mkdtemp', return_value=REPO_DIR)
+@patch('sagemaker.git_utils._validate_git_config')
+@patch('sagemaker.git_utils._checkout_branch_and_commit')
```

Comment:
You should not mock ```_checkout_branch_and_commit```, you should modules that this private functions calls instead. You should mock ```subprocess.check_call``` here and in other tests in this file.